storageframework/src/vespa/storageframework/generic/thread/tickingthread.cpp

```diff
@@ -120,40 +124,40 @@ class TickingThreadPoolImpl final : public TickingThreadPool {
         void broadcast() override {}
     };
     struct CriticalGuard final : public TickingLockGuard::Impl {
-        vespalib::MonitorGuard _guard;
+        std::condition_variable &_cond;
 
-        explicit CriticalGuard(vespalib::Monitor& m) : _guard(m) {}
+        explicit CriticalGuard(std::condition_variable & cond) : _cond(cond) {}
```

Comment:
Shouldn't this class take in a `mutex` and a `condition_variable` and instantiate a `unique_lock` instead? The previous impl acquired a `MonitorGuard` while this impl does not end up taking any locks.

Reply:
Oops, to fast....

Reply:
Fixed